src/Tasks/GenerateResource.cs

```diff
@@ -2432,6 +2432,12 @@ internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
                     {
                         if (String.Compare(requestedAssemblyName.Name, candidateAssemblyName.Name, StringComparison.CurrentCultureIgnoreCase) == 0)
                         {
+                            string itemSpec = _assemblyFiles[i].ItemSpec;
+
+                            if (requestedAssemblyName.Name.Contains("System.Runtime"))
```

Comment:
@ericstj 

I was taking a look at doing resource extraction via `System.Reflection.Metadata` (rather than loading the assembly, as you suggested), and I’m pretty baffled. Do you know of a good contact that I could bug to see if what I want to do is possible?

The MSDN docs seem to just be the doc-comments, with no examples or motivations, and the only use I could find was in Roslyn, which is write-only.

I can successfully extract all the names of the embedded resources

The existing reflection-based code:
1.	[Loads the assembly](https://github.com/Microsoft/msbuild/blob/00ff8ba81a8046f87a4476d5f14113e0088c5ca4/src/Tasks/GenerateResource.cs#L2958) (this failing when loading an ARM assembly is what I really want to fix)
2.	[Gets a list of resource names](https://github.com/Microsoft/msbuild/blob/00ff8ba81a8046f87a4476d5f14113e0088c5ca4/src/Tasks/GenerateResource.cs#L3013)
3.	[Extracts each resource name from its stream](https://github.com/Microsoft/msbuild/blob/00ff8ba81a8046f87a4476d5f14113e0088c5ca4/src/Tasks/GenerateResource.cs#L3050-L3085) and saves it for extraction.

I can do steps 1 and 2 fairly reasonably:
```c#
using (var stream = new FileStream(args[0], FileMode.Open, FileAccess.Read,
    FileShare.Read | FileShare.Delete))
using (var peReader = new PEReader(stream))
{
    var r = peReader.GetMetadataReader();
    foreach (var resourceHandle in r.ManifestResources)
    {
        var resource = r.GetManifestResource(resourceHandle);
        if (resource.Implementation.IsNil)
        {
            string resName = r.GetString(resource.Name);

```

But I can't see how to translate a `System.Reflection.Metadata.ManifestResource` into a `Stream` suitable for passing into a `ResourceReader`. It has fields for `Implementation` (Nil in my simple test program) and `Offset` (0 in my simple test). I don't know how to turn those into bytes.